examples/basic_gpio_example.py

```diff
@@ -0,0 +1,111 @@
+    """
+    device = SupernovaDevice()
+
+    print("Opening Supernova host adapter device and getting access to the GPIO interface...")
+    info = device.open()
+    gpio = device.create_interface("gpio")
+
+    print("Initializing GPIO peripheral.")
+    (success, _) = gpio.set_bus_voltage(3300)
+    if not success:
+        print("Couldn't set the GPIO bus voltage.")
+        exit(1)
+
+    print("Configuring GPIO pins...")
+    (success, _) = gpio.configure_pin(GpioPinNumber.GPIO_6, GpioFunctionality.DIGITAL_OUTPUT)
+    if not success:
+        print("Couldn't configure GPIO_6 as Digital Output.")
+        exit(1)
+
+    (success, _) = gpio.configure_pin(GpioPinNumber.GPIO_5, GpioFunctionality.DIGITAL_INPUT)
+    if not success:
+        print("Couldn't configure GPIO_5 as Digital Input.")
+        exit(1)
+
+    print("Setting GPIO_6 to HIGH and reading value from GPIO_5...")
+    (success, _) = gpio.digital_write(GpioPinNumber.GPIO_6, GpioLogicLevel.HIGH)
+    if not success:
+        print("Couldn't set GPIO_6 to HIGH.")
+        exit(1)
+
+    (success, value) = gpio.digital_read(GpioPinNumber.GPIO_5)
+    if success:
+        print(f"GPIO_5 read value: {value}")
+    else:
+        print("Failed to read GPIO_5 value.")
+
+    print("Setting GPIO_6 to LOW and reading value from GPIO_5...")
+    (success, _) = gpio.digital_write(GpioPinNumber.GPIO_6, GpioLogicLevel.LOW)
+    if not success:
+        print("Couldn't set GPIO_6 to LOW.")
+        exit(1)
+
+    (success, value) = gpio.digital_read(GpioPinNumber.GPIO_5)
+    if success:
+        print(f"GPIO_5 read value: {value}")
+    else:
+        print("Failed to read GPIO_5 value.")
+
+    # -- Interruptuions
+    def is_gpio_interrupt(name, message):
+        is_gpio_interruption = message['name'].strip() == "GPIO INTERRUPTION"
+        return message['name'].strip() == "GPIO INTERRUPTION"
+    
+    def handle_gpio_interrupt(name, message):
+        print(f"GPIO interrupt notification on pin: {message['pin_number']}")
+        gpio_interrupt_event.set()
+
+    print("Registering GPIO interrupt handlers...")
+    device.on_notification(name="GPIO INTERRUPTION", filter_func=is_gpio_interrupt, handler_func=handle_gpio_interrupt) # gpio-interruption
+
+    print("Configuring interrupt on GPIO_5 for both rising and falling edges...")
+    (success, _) = gpio.set_interrupt(GpioPinNumber.GPIO_5, GpioTriggerType.TRIGGER_BOTH_EDGES)
+    if not success:
+        print("Couldn't set interrupt on GPIO_5.")
+        exit(1)
+
+    print("Toggling GPIO_6 to generate interrupts on GPIO_5...")
+    for level in [GpioLogicLevel.HIGH, GpioLogicLevel.LOW, GpioLogicLevel.HIGH, GpioLogicLevel.LOW]:
+        (success, _) = gpio.digital_write(GpioPinNumber.GPIO_6, level)
+        if not success:
+            print(f"Couldn't set GPIO_6 to {level}.")
+            exit(1)
+        # Wait for the GPIO interrupt to be processed
+        gpio_interrupt_event.wait()
+        gpio_interrupt_event.clear()
```

Comment:
Maybe we could be explicit on how many interruptions we are triggering/expecting